app/javascript/dashboard/store/modules/inboxes.js

```diff
@@ -76,10 +77,14 @@ export const actions = {
       throw new Error(error);
     }
   },
-  updateAutoAssignment: async (nil, { id, ...inboxParams }) => {
+  updateAutoAssignment: async ({ commit }, { id, ...inboxParams }) => {
+    commit(types.default.SET_INBOXES_UI_FLAG, { isUpdatingAutoAssignment: true });
     try {
-      await InboxesAPI.update(id, inboxParams);
+      const response = await InboxesAPI.update(id, inboxParams);
+      commit(types.default.EDIT_INBOXES, response.data);
+      commit(types.default.SET_INBOXES_UI_FLAG, { isUpdatingAutoAssignment: false });
```

Comment:
Please update the test cases for the same :)

Reply:
Yes, I am currently doing this :)